### PR TITLE
Fix Open XL compiler warnings

### PR DESCRIPTION
**Pull request by Deigue**
Changes along with the diagnostic flags:

compiler/optimizer/ValuePropagationCommon.cpp [-Wsometimes-uninitialized]
include_core/unix/zos/edcwccwi.h [-Wtrigraphs]
util/a2e/atoe.c [-Wwritable-strings]
util/a2e/headers/stdio.h [-Wwritable-strings]

### CODE REVIEW

**Comment by Deigue** [comment]
@keithc-ca I'm opening the file `include_core/unix/zos/edcwccwi.h` again reviewing that there was line ending issue pointed out, and I am seeing trailing whitespaces pretty much on the entire file. 
Just confirming, but do I trim the whitespace for the entire file, or there is some reason why it was done like this in the first place?

**Comment by keithc-ca** [comment]
I believe the check only considers lines modified in this pull request, but I would support removing trailing whitespace from all lines.

**Comment by Deigue** [comment]
Removed all the trailing whitespace from `include_core/unix/zos/edcwccwi.h`

**Comment by babsingh** [comment]
jenkins build all

File: compiler/optimizer/ValuePropagationCommon.cpp [hunk 1]
```diff
@@ -3295,7 +3295,7 @@ void OMR::ValuePropagation::transformRealTimeArrayCopy(TR_RealTimeArrayCopy *rtA
    //create ifNodes
    TR::Node *spineShiftNode = TR::Node::create(vcallNode, TR::iconst, 0, (int32_t)fe()->getArraySpineShift(elementSize));
    TR::TreeTop *ifTree;
-   TR::Node *ifNode;
+   TR::Node *ifNode = nullptr;
```
Comment (babsingh) [inline compiler/optimizer/ValuePropagationCommon.cpp:3298]: For consistency, can it be initialized similar to the adjacent local variables?
```suggestion
   TR::Node *ifNode = NULL;
```

Comment (babsingh) [reply compiler/optimizer/ValuePropagationCommon.cpp:3298]: In `ValuePropagationCommon.cpp`, there are many other uninitialized variables. Can you initialize all of them?

**Comment by babsingh** [comment]
jenkins build all

**Review by babsingh — APPROVED** [review]
LGTM. Ignoring the AIX failure: all AIX nodes are offline.

**Pull request merged**